cosmos/dbt/selector.py

```diff
@@ -85,6 +85,9 @@ def select_nodes_ids_by_intersection(nodes: dict[str, DbtNode], config: Selector
     """
     selected_nodes = set()
     for node_id, node in nodes.items():
+        if not config.paths and not config.tags and not config.config:
```

Comment:
@jensenity What do you think of bringing this statement to the top of the method and using it as a condition before entering the `for`  loop? This would reduce the cyclomatic complexity from O(n) to O(1).

What do you think if we created a method in `SelectorConfig`, `is_empty`, which would return (`config.paths and config.tags and config.config`)?

Also, which select statement led to this uncovering this issue in production?

Reply:
[0200aef](https://github.com/astronomer/astronomer-cosmos/pull/463/commits/0200aef4cc99c8a8d17bbf0d2f60583e8764bfa7) @tatiana is this better?

Reply:
It is, @jensenity , thank you! I think that the inclusion of `self.other` is breaking tests

Reply:
>Also, which select statement led to this uncovering this issue in production?

So I discovered this when I was testing the exclude test case. And when I ran the test case, for instance, if the test case is only selecting "is_child"..It doesnt not return on the child node, however, the grandparent & parent is all return
<img width="1052" alt="Screen Shot 2023-08-15 at 8 57 28 PM" src="https://github.com/astronomer/astronomer-cosmos/assets/39326969/96d83d77-9510-4f13-8d48-46b208739e4a">

`test_select_nodes_by_select_union_config_tag`
And for this test, it returns all 5 nodes. Thats how i discovered it.

Reply:
Thanks, @jensenity ! Happy to merge once the tests are passing.

Reply:
[9985698](https://github.com/astronomer/astronomer-cosmos/pull/463/commits/998569811e2f2ea8e964b0f94823e05f414d971e)